### ISSUE TITLE
Fix SQL runner in demo (without running queries)

### ISSUE THREAD

**Issue by rephus** [lightdash/lightdash#12169]
### Description

tl;dr Allow users to access `sql runner` but don't allow `running queries` 

Most of the endpoints are not available on the page 

![image](https://github.com/user-attachments/assets/12713205-6181-4d1c-b1ee-a489919657dd)

And also, github is also throwing an error (write back feature) 
THis is not blocking the user from using the sql runner, but we should not throw an error here: 

![image](https://github.com/user-attachments/assets/6ca75d62-345f-4e21-81ee-f2fbe0fe000d)

### Steps to Reproduce the Bug or Issue

https://demo.lightdash.com/projects/2014e038-ff4b-4761-ae6f-fbf551e7b468/sql-runner

### Cloud or self-hosting

None
